Tennis-scheduler/src/main/java/com/tennis/tennisscheduler/controllers/PersonController.java

```diff
@@ -59,13 +64,25 @@ public ResponseEntity<PersonDto> getPersonById(@PathVariable long id){
         return new ResponseEntity<>(personDtoMapper.fromPersonToPersonDto(person),HttpStatus.OK);
     }
     @PutMapping("/{id}")
+    @PreAuthorize("hasAnyRole('TENNIS_PLAYER','ADMIN')")
     public ResponseEntity<PersonDto> updatePerson(@PathVariable long id,@RequestBody PersonDto personDto){
+
         Person personExisting = personService.findById(id);
-        if (personExisting == null) {
```

Comment:
You can leave this check here. No matter the role, if person doesn't exist in the database nothing else should be done. Also return NOT_FOUND since that is what happened, BAD_REQUEST means that request is malformed.